app/controlplane/internal/service/auth.go

```diff
@@ -289,6 +301,61 @@ func callbackHandler(svc *AuthService, w http.ResponseWriter, r *http.Request) (
 	return http.StatusTemporaryRedirect, nil
 }
 
+// autoOnboardOnOrganizations onboards the user to the organizations specified in the config
+// If the organization does not exist, it will be created with an inline CAS backend
+// If the user is not already a member of the organization, a membership will be created with the role specified in the config
+func autoOnboardOnOrganizations(ctx context.Context, svc *AuthService, u *biz.User) error {
+	for _, spec := range svc.onboardingConfig {
+		// Ensure the organization exists or create it if it doesn't
+		org, err := svc.orgUseCase.FindOrCreate(ctx, spec.GetName())
```

Comment:
You might need to introduce some locking here. This code might be run by different threads or pods, so it must be reentrant.